kratos/includes/global_variables.h

```diff
@@ -34,6 +34,9 @@ namespace Kratos
 namespace Globals
 {
 	constexpr double Pi   = 3.14159265358979323846264338327950288419716939937510582097494459230781640628620899862803482534211706798214808651L;
+
+    static constexpr int MaxAllowedThreads = 128; //we assume that no more than MaxAllowedThreads is used in SMP
```

Comment:
i think this should be a general constant. moving to another archive would be of course trivial, however leaving it here would allow to use it consistently...

Reply:
what would you use it for? I mean why limit it?

Reply:
This is used as a size for the statically sized accumulator array used in reduction. Without the limit, we would have to dynamically allocate the array, which is expensive.